src/Cli.Tests/EndToEndTests.cs

```diff
@@ -668,7 +668,12 @@ public void TestHelpWriterOutput(string command, string flags, string[] expected
         Assert.IsNotNull(output);
 
         // Version Info logged by dab irrespective of commands being parsed correctly.
-        StringAssert.Contains(output, $"{Program.PRODUCT_NAME} {ProductInfo.GetProductVersion()}", StringComparison.Ordinal);
+        StringAssert.Contains(ProductInfo.GetProductVersion(), $"{Program.PRODUCT_NAME} {ProductInfo.GetProductVersion()}", StringComparison.Ordinal);
+
+        // Check that build hash is returned as part of  version number
+        string[] versionParts = output.Split('+');
+        Assert.AreEqual(2, versionParts.Length, "Build hash not returned as part of version number.");
+        Assert.AreEqual(41, versionParts[1].Length, "Build hash is not of expected length.");
```

Comment:
The `output` here could be different based on the command used in the test case. If you look at the various DataRows specified as a function header, only the first entry about version will have the second part to be exactly equal to the hash length.

The same wouldn't hold true for other cases. For example, for the test case: `edit --new-option` the output seen would be: 

> Microsoft.DataApiBuilder 0.9.6-rc+529471ba3b50e617439c57548891fdff7ec1c3fa
> c Microsoft Corporation. All rights reserved.
> 
> ERROR(S):
>   Verb 'edit' is not recognized.
> 
>   --help       Display this help screen.
> 
>   --version    Display version information.

So, we should not assert on the hash length - it will always be a subset of the whole output. Same for the versionParts.Length, there could be more than one `+` in the entire output

Reply:
if you just want to verify ProductVersion contains the hash, we can simply do that with unit test asserts on the `ProductInfo.GetProductVersion()` - that it contains the +, and hash is of right length. This doesnt need to be added in this test: `TestVersionInfoAndConfigIsCorrectlyDisplayedWithDifferentCommand` - you can define a separate unit test method altogether as well

Reply:
Make sense. I created a new `TestVersionHasBuildHash` test